### PR TITLE
Support python 3.6

### LINKED ISSUE #22: Add support for python 3.6
We'd like to be able to use duet in python3.6 since it's still used by colab and internally at google (see https://github.com/quantumlib/Cirq/pull/4009). Among other things, this will require adding back in support for AsyncContextManager and adding a 3.6-only dependency on [contextvars](https://pypi.org/project/contextvars/). There may be some other changes required as well.

### PR DESCRIPTION
**Pull request by fedimser**
To make it work with python 3.6:
1. Don't use postponed evaluation of annotations (added in python 3.7). For that had to put some type annotations in quotes.
2. Use contextvars library from pip (rather than from standard library) for Python 3.6.
3. Use asynccontextmanager from pip library [asyncio-contextmanager](https://pypi.org/project/asyncio-contextmanager/) rather than from standard library's contextlib. Alternative would be vendoring it from standard library ([source](https://github.com/python/cpython/blob/3.9/Lib/contextlib.py)). Implementation in pip library is almost equivalent to the one from standard library - there are only small discrepancies in exception handling and error messages.

Added requirements file and modified CI to run tests for Python 3.6.

Fixes #22.

### CODE REVIEW

**Review by maffoo — COMMENTED** [review]
Nice! Just a few minor comments.

File: .gitignore [hunk 1]
```diff
@@ -112,6 +112,7 @@ venv.bak/
 
 # IDEs
 *.iml
+.idea
```
[the remark this answers is not in the document]
Comment (fedimser) [reply .gitignore]: Done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -59,14 +59,16 @@ jobs:
           python-version: ${{ matrix.python-version }}
           architecture: 'x64'
       - name: Install requirements
-        run: pip install -r dev/requirements.txt
+        run: |
+          pip install -r requirements.txt
+          pip install -r dev/requirements.txt
```
[the remark this answers is not in the document]
Comment (fedimser) [reply .github/workflows/ci.yml]: Done

File: duet/__init__.py [hunk 3]
```diff
@@ -61,6 +59,15 @@
 T = TypeVar("T")
 U = TypeVar("U")
 
+try:
+    asynccontextmanager = contextlib.asynccontextmanager
+except AttributeError:
+    # In python 3.6 asynccontextmanager isn't available from the standard library, so we are using
+    # equivalent third-party implementation.
+    from aiocontext import async_contextmanager  # type: ignore
```
[the remark this answers is not in the document]
Comment (fedimser) [reply duet/__init__.py]: Done

**Review by maffoo — APPROVED** [review]
LGTM!

**Pull request merged**